CMakeLists.txt

```diff
@@ -820,6 +825,24 @@ check_symbol_exists(setsockopt     "${CURL_INCLUDES}" HAVE_SETSOCKOPT)
 # symbol exists in win32, but function does not.
 check_function_exists(inet_pton HAVE_INET_PTON)
 
+if(CURL_XATTR)
+  check_symbol_exists(fsetxattr "${CURL_INCLUDES}" HAVE_FSETXATTR)
+  if(HAVE_FSETXATTR)
+    add_definitions(-DHAVE_FSETXATTR)
+    foreach(CURL_TEST HAVE_FSETXATTR_5 HAVE_FSETXATTR_6)
+      curl_internal_test_run(${CURL_TEST})
+    endforeach(CURL_TEST)
+    if(HAVE_FSETXATTR_5)
+      set_source_files_properties(tool_xattr.c PROPERTIES
+        COMPILE_FLAGS "-DHAVE_FSETXATTR_5")
```

Comment:
This can possibly override other compile flags, so try appending properties and use the special property `COMPILE_DEFINITIONS`:

    set_property(SOURCE tool_xattr.c APPEND PROPERTY
      COMPILE_DEFINITIONS HAVE_FSETXATTR_5)

Same comment for the next line. Alternatively, try adding a `#cmakedefine` as before, autotools uses an `AC_DEFINE` for these `..._5` and `..._6` options too.